src/array/from.rs

```diff
@@ -85,14 +85,6 @@ impl<T: DaftDataType> TryFrom<(&str, Box<dyn arrow2::array::Array>)> for DataArr
 
     fn try_from(item: (&str, Box<dyn arrow2::array::Array>)) -> DaftResult<Self> {
```

Comment:
`DataArray::new()` actually returns a `DaftResult` itself, where an `Err()` is returned if the field type doesn't match the array type, so this would still be a `TryFrom` implementation.

However, I _am_ slightly concerned that we're not doing any runtime checking of the supplied `Box<dyn arrow2::array::Array>>` dtype against the `T` dtype, since it seems as if one could do something like:
```rust
let arr = Box::new(Int32Array::from(&[Some(1), None, Some(3)]));

let daft_arr_1 = DataArray::<Int32Array>::try_from(("foo", arr));
let daft_arr_2 = DataArray::<Utf8Array>::try_from(("foo", arr));
```
and I don't think that this would fail. Not sure how big of a concern this is.